sonarr/downloadclient.go

```diff
@@ -0,0 +1,143 @@
+package sonarr
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"path"
+
+	"golift.io/starr"
+)
+
+// Define Base Path for downloadClient calls.
+const bpDownloadClient = APIver + "/downloadClient"
+
+// DownloadClientInput is the input for a new or updated downladClient.
+type DownloadClientInput struct {
+	Enable                   bool                `json:"enable"`
+	RemoveCompletedDownloads bool                `json:"removeCompletedDownloads"`
+	RemoveFailedDownloads    bool                `json:"removeFailedDownloads"`
+	Priority                 int                 `json:"priority"`
+	ID                       int64               `json:"id,omitempty"`
+	ConfigContract           string              `json:"configContract"`
+	Implementation           string              `json:"implementation"`
+	Name                     string              `json:"name"`
+	Protocol                 string              `json:"protocol"`
+	Tags                     []int               `json:"tags"`
+	Fields                   []*starr.FieldInput `json:"fields"`
+}
+
+// DownloadClientOutput is the output from the downladClient methods.
+type DownloadClientOutput struct {
+	Enable                   bool                 `json:"enable"`
+	RemoveCompletedDownloads bool                 `json:"removeCompletedDownloads"`
+	RemoveFailedDownloads    bool                 `json:"removeFailedDownloads"`
+	Priority                 int                  `json:"priority"`
+	ID                       int64                `json:"id,omitempty"`
+	ConfigContract           string               `json:"configContract"`
+	Implementation           string               `json:"implementation"`
+	ImplementationName       string               `json:"implementationName"`
+	InfoLink                 string               `json:"infoLink"`
+	Name                     string               `json:"name"`
+	Protocol                 string               `json:"protocol"`
+	Tags                     []int                `json:"tags"`
+	Fields                   []*starr.FieldOutput `json:"fields"`
+}
+
+// GetDownloadClients returns all configured downloadclients.
+func (s *Sonarr) GetDownloadClients() ([]*DownloadClientOutput, error) {
+	return s.GetDownloadClientsContext(context.Background())
+}
+
+// GetDownloadClientsContext returns all configured downloadclients.
+func (s *Sonarr) GetDownloadClientsContext(ctx context.Context) ([]*DownloadClientOutput, error) {
+	var output []*DownloadClientOutput
+
+	req := starr.Request{URI: bpDownloadClient}
+	if err := s.GetInto(ctx, req, &output); err != nil {
+		return nil, fmt.Errorf("api.Get%s): %w", &req, err)
+	}
+
+	return output, nil
+}
+
+// GetDownloadClient returns a single downloadclient.
```

Comment:
"download client."

Reply:
ok, then we'll use this standard for comments. lowercase with spaces

Reply:
Yes, please. You can write method or type names in comments without spaces, but like this it looks better with a space.